File: python/q0209/solution.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        left, right = 0, 0
        ans = len(nums) + 1
        cur_sum = nums[0]
        while left <= right < len(nums):
            if cur_sum < target:
                right += 1
                if right == len(nums):
                    break
                cur_sum += nums[right]
            else:
                ans = min(ans, right - left + 1)
                cur_sum -= nums[left]
                left += 1
        return 0 if ans == len(nums) + 1 else ans


sol = Solution()
ans = sol.minSubArrayLen(4, [1,1,1])
print(ans)

File: java/src/q0209/Solution.java
package q0209;

class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0 ;
        int right = 0;
        int ans = Integer.MAX_VALUE;
        int sum = 0;
        while (right < nums.length) {
            sum += nums[right];
            while (sum >= target) {
                ans = Math.min(ans, right - left + 1);
                sum -= nums[left++];
            }
            right++;
        }
        return ans == Integer.MAX_VALUE ? 0 : ans;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int target = 7;
        int[] nums = {2,3,1,2,4,3};
        int res = solution.minSubArrayLen(target, nums);
        System.out.println(res);
    }
}

